src/Libraries/Microsoft.Extensions.TimeProvider.Testing/FakeTimeProvider.cs

```diff
@@ -158,28 +159,43 @@ internal void RemoveWaiter(FakeTimeProviderTimer.Waiter waiter)
         }
     }
 
-    private List<FakeTimeProviderTimer.Waiter> GetWaitersToWake()
+    private FakeTimeProviderTimer.Waiter? TryGetWaiterToWake(DateTimeOffset targetNow)
     {
-        var l = new List<FakeTimeProviderTimer.Waiter>(Waiters.Count);
-        foreach (var w in Waiters)
+        var candidate = default(FakeTimeProviderTimer.Waiter);
+
+        lock (Waiters)
         {
-            if (_now >= w.WakeupTime)
+            if (Waiters.Count == 0)
             {
-                l.Add(w);
+                return null;
             }
-        }
 
-        return l;
-    }
-
-    private void WakeWaiters(List<FakeTimeProviderTimer.Waiter> waiters)
-    {
-        foreach (var w in waiters)
-        {
-            if (_now >= w.WakeupTime)
+            // Search through Waiters and find the waiter with the minimum
+            // WakeupTime which is less than or equal to targetNow.
```

Comment:
This comment appears to reiterate what the code below does. Such are generally not that useful. Useful comments explain "why" something is done.

Reply:
I agree, good point.